### PR TITLE
Print warning for deprecated properties in build and bundle command 

### PR DESCRIPTION
**Pull request by troydai**
https://github.com/aspnet/XRE/issues/1292

### CODE REVIEW

**Comment by troydai** [comment]
@davidfowl @ChengTian

**Comment by troydai** [comment]
DTH will be handle in another PR since the scope is larger.

**Comment by BrennanConroy** [comment]
What happened to fixing the misspelling of occurred

**Comment by troydai** [comment]
All right, all right, I'll revert it.

File: src/Microsoft.Framework.PackageManager/Building/BuildManager.cs [hunk 1]
```diff
@@ -33,13 +33,22 @@ public BuildManager(IServiceProvider hostServices, BuildOptions buildOptions)
...
+            var warnings = new List<string>();
             Runtime.Project project;
-            if (!Runtime.Project.TryGetProject(_buildOptions.ProjectDir, out project))
+            if (!Runtime.Project.TryGetProject(_buildOptions.ProjectDir, out project, warnings))
             {
                 WriteError(string.Format("Unable to locate {0}.", Runtime.Project.ProjectFileName));
                 return false;
             }
 
+            if (warnings.Any())
```
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Building/BuildManager.cs]: If check shouldn't be necessary

**Comment by troydai** [comment]
@muratg @davidfowl @anurse @victorhurdugaci

File: src/Microsoft.Framework.Runtime.Hosting/ProjectFilesCollection.cs [hunk 2]
```diff
@@ -39,20 +39,20 @@ public ProjectFilesCollection(JObject rawProject, string projectDirectory, strin
 
             var bundleExcludePatterns = PatternsCollectionHelper.GetPatternsCollection(rawProject, projectDirectory, projectFilePath, "bundleExclude", DefaultBundleExcludePatterns);
 
-            _sharedPatternsGroup = PatternGroup.Build(rawProject, projectDirectory, projectFilePath, "shared", legacyName: null, fallbackIncluding: DefaultSharedPatterns, additionalExcluding: excludePatterns);
+            _sharedPatternsGroup = PatternGroup.Build(rawProject, projectDirectory, projectFilePath, "shared", legacyName: null, warnings: warnings, fallbackIncluding: DefaultSharedPatterns, additionalExcluding: excludePatterns);
 
-            _resourcePatternsGroup = PatternGroup.Build(rawProject, projectDirectory, projectFilePath, "resource", "resources", fallbackIncluding: DefaultResourcesPatterns, additionalExcluding: excludePatterns);
+            _resourcePatternsGroup = PatternGroup.Build(rawProject, projectDirectory, projectFilePath, "resource", "resources", warnings: warnings, fallbackIncluding: DefaultResourcesPatterns, additionalExcluding: excludePatterns);
```
Comment (ChengTian) [inline src/Microsoft.Framework.Runtime.Hosting/ProjectFilesCollection.cs]: `legacyName: "resources"`, same for lines below

**Comment by ChengTian** [comment]
Add TODO comments saying that legacy names should retire later. Then :shipit:

**Comment by muratg** [comment]
In addition to @ChengTian's comment, please also file a bug to track the work to remove legacy names later.

**Comment by troydai** [comment]
Add bug https://github.com/aspnet/DNX/issues/1379

**Pull request merged**